microsoft/testsuites/core/timesync.py

```diff
@@ -151,3 +153,74 @@ def timesync_check_clocksource(self, node: Node) -> None:
+            1. Current clock event name is 'Hyper-V clockevent'.
+            2. 'Hyper-V clockevent' and 'hrtimer_interrupt' show up times in
+             /proc/timer_list should equal to cpu count.
+            3. when cpu count is 1 and cpu type is not AMD type, unbind current time
+             clock event, check current time clock event switch to 'lapic'.
+        """,
+        priority=1,
+    )
+    def timesync_check_unbind_clockevent(self, node: Node) -> None:
+        if node.shell.exists(PurePosixPath(f"{self.current_clockevent}")):
+            # 1. Current clock event name is 'Hyper-V clockevent'.
+            clock_event_name = "Hyper-V clockevent"
+            event_handler_name = "hrtimer_interrupt"
+            cat = node.tools[Cat]
+            clock_event_result = cat.run(self.current_clockevent)
+            assert_that(clock_event_result.stdout).described_as(
+                f"Expected clockevent name is {clock_event_name}, "
+                f"but actual it is {clock_event_result.stdout}."
+            ).is_equal_to(clock_event_name)
+
+            # 2. 'Hyper-V clockevent' and 'hrtimer_interrupt' show up times in
+            #  /proc/timer_list should equal to cpu count.
+            timer_list_result = cat.run("/proc/timer_list", sudo=True)
+            lscpu = node.tools[Lscpu]
+            core_count = lscpu.get_core_count()
+            event_handler_times = timer_list_result.stdout.count(
+                f"{event_handler_name}"
+            )
+            assert_that(event_handler_times).described_as(
+                f"Expected {event_handler_name} shown up {core_count} times in output "
+                f"of /proc/timer_list, but actual it shows up "
+                f"{event_handler_times} times."
+            ).is_equal_to(core_count)
+
+            clock_event_times = timer_list_result.stdout.count(f"{clock_event_name}")
+            assert_that(clock_event_times).described_as(
+                f"Expected {clock_event_name} shown up {core_count} times in output "
+                f"of /proc/timer_list, but actual it shows up "
+                f"{clock_event_times} times."
+            ).is_equal_to(core_count)
+
+            # 3. when cpu count is 1 and cpu type is not AMD type, unbind current time
+            #  clock event, check current time clock event switch to 'lapic'.
+            if not lscpu.is_amd_cpu() and 1 == core_count:
+                cmd_result = node.execute(
+                    f"echo {clock_event_name} > {self.unbind_clockevent}",
+                    sudo=True,
+                    shell=True,
+                )
+                assert_that(cmd_result.exit_code).described_as(
+                    f"Fail to execute command "
+                    f"[echo {clock_event_name} > {self.unbind_clockevent}]."
+                ).is_equal_to(0)
+                timout_timer = create_timer()
+                timeout = 30
+                while timout_timer.elapsed(False) < timeout:
```

Comment:
It looks this logic is in another test case, use a method to avoid copy/paste code.

Reply:
Or encapsulate it under the already used "retry" module